server/src/test/java/org/elasticsearch/action/ActionModuleTests.java

```diff
@@ -194,4 +198,47 @@ public void handleRequest(RestRequest request, RestChannel channel, NodeClient c
             threadPool.shutdown();
         }
     }
+
+    public void test3rdPartyHandlerIsNotInstalled() {
+        Settings settings = Settings.builder()
+            .put("xpack.security.enabled", false)
+            .put("path.home", createTempDir())
+            .build();
+
+        SettingsModule settingsModule = new SettingsModule(Settings.EMPTY);
+        ThreadPool threadPool = new TestThreadPool(getTestName());
+        ActionPlugin secPlugin = new SecPlugin();
+        try {
+            UsageService usageService = new UsageService();
+
+            Exception e = expectThrows(IllegalArgumentException.class, () ->
+                new ActionModule(settingsModule.getSettings(),
+                    TestIndexNameExpressionResolver.newInstance(threadPool.getThreadContext()),
+                    settingsModule.getIndexScopedSettings(), settingsModule.getClusterSettings(), settingsModule.getSettingsFilter(),
+                    threadPool, Arrays.asList(secPlugin), null, null, usageService, null)
+            );
+            assertThat(e.getMessage(), Matchers.equalTo("The org.elasticsearch.action.ActionModuleTests$SecPlugin plugin tried to " +
+                "install a custom REST wrapper. This functionality is not available anymore."));
+        } finally {
+            threadPool.shutdown();
+        }
+    }
+
+    class FakeHandler implements RestHandler {
+        @Override
+        public List<Route> routes() {
+            return singletonList(new Route(GET, "/_dummy"));
+        }
+
+        @Override
+        public void handleRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
+        }
+    }
+
+    class SecPlugin implements ActionPlugin {
+        @Override
+        public UnaryOperator<RestHandler> getRestHandlerWrapper(ThreadContext threadContext) {
+            return handler -> new FakeHandler();
```

Comment:
This is fine - there's no need to change it, but it would have been fine to just make this a no-op operator.
```suggestion
            return UnaryOperator.identity();
```
We really only care that the plugin returns _something_. We don't care what it does.